rq/worker.py

```diff
@@ -621,9 +621,20 @@ def monitor_work_horse(self, job):
                 # which we don't want to catch, so we re-raise those ones.
                 if e.errno != errno.EINTR:
                     raise
+                # Send a heartbeat to keep the worker alive.
+                self.heartbeat()
 
     def _monitor_work_horse_tick(self, job):
-        _, ret_val = os.waitpid(self._horse_pid, 0)
+        while True:
+            try:
+                with UnixSignalDeathPenalty(self.default_worker_ttl - 15, HorseMonitorTimeoutException):
```

Comment:
I think we can hard code this so that heartbeat is always sent every 30 seconds, when the worker is performing a job. Worker keys can then be set to always expire in 45 seconds when job is under way so we'll know that a worker is dead in less than 1 minute's time.

What do you think?

Reply:
Generally, I'm ok with that.
However, the expiration timeout is set in:
`heartbeat()` with `timeout = max(timeout, self.default_worker_ttl)`. 
420 is always bigger than 45.

I'm not sure how I should change this without causing havoc in the rest of the worker...
We could introduce a new `DEFAULT_WORKER_BUSY_TTL = 45` and make it more explicit which timeout should be used when.

Reply:
I pushed a new commit with a change to the heartbeat timeout.